### PR TITLE
[docs] fix links in location docs

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Links and formatting in location docs are broken.

# How

- Fixed links by removing `expo` from hashtags
- Removed backticks inside and around links. They didn't look like links and in some cases they caused formatting issues.

# Test Plan

We will see once we deploy docs to staging 🤷‍♂️

### CODE REVIEW

**Comment by nicknovitski** [comment]
Ah, I see what you mean, the way we format code elements inside links (such as ``[`foo`](bar)``) looks bad.

I changed the formatting of all argument type documentation to have backticks, because mdx  interprets types like `Array<any>` as having an opening tag for an `any` element.  But I've realized now this was a mistake, I should have just changed them to escape the angle-brackets directly, like `Array\<any\>`.  I should make that change.

In the meantime, this change is a good idea.  I've checked it locally, it does look better.

File: docs/pages/versions/v32.0.0/sdk/location.md [hunk 1]
```diff
@@ -45,12 +45,12 @@ Returns a promise resolving to an object representing [Location](#typelocation)
...
 
-Subscribe to location updates from the device. Please note that updates will only occur while the application is in the foreground. To get location updates while in background you'll need to use [`Location.startLocationUpdatesAsync`](#expolocationstartlocationupdatesasync).
+Subscribe to location updates from the device. Please note that updates will only occur while the application is in the foreground. To get location updates while in background you'll need to use [Location.startLocationUpdatesAsync](#locationstartlocationupdatesasync).
 
 #### Arguments
 
 -   **options : `object`** -- A map of options:
-    -   **accuracy (_[Location.Accuracy](#expolocationaccuracy)_** -- Location manager accuracy. Pass one of [Location.Accuracy](#expolocationaccuracy) enum values. For low accuracy the implementation can avoid geolocation providers that consume a significant amount of power (such as GPS).
+    -   **accuracy (_[Location.Accuracy](#locationaccuracy)_** -- Location manager accuracy. Pass one of [Location.Accuracy](#locationaccuracy) enum values. For low accuracy the implementation can avoid geolocation providers that consume a significant amount of power (such as GPS).
```
Comment (nicknovitski) [inline docs/pages/versions/v32.0.0/sdk/location.md]:
```suggestion
    -   **accuracy (_[Location.Accuracy](#locationaccuracy)_)** -- Location manager accuracy. Pass one of [Location.Accuracy](#locationaccuracy) enum values. For low accuracy the implementation can avoid geolocation providers that consume a significant amount of power (such as GPS).
```

This is what type information used to look like in the docs: `- **argument/key (_[Expo.Whatever](#expowhatever)_)**`

**Review by sjchmiela — COMMENTED** [review]
Couldn't we use instead of
```
`[Location.Accuracy](#locationaccuracy)`
```
:
```
[`Location.Accuracy`](#locationaccuracy)
```
?

**Comment by tsapeta** [comment]
@sjchmiela I'd say yes, but
```
[`Location.Accuracy`](#locationaccuracy)
```
is styled as a code block so you don't know it's a link until you hover on it (bad UX). Ofc, we can change the styling, but I guess it shouldn't be in this PR.

**Pull request merged**